hypothesis-python/src/hypothesis/extra/_array_helpers.py

```diff
@@ -673,7 +685,7 @@ def do_draw(self, data):
             while j < len(result) and result[j] == slice(None):
                 j += 1
             result[i:j] = [Ellipsis]
-        else:
+        elif self.allow_fewer_indices_than_dims:
```

Comment:
We don't actually have a test where this is `False`, so it might be time to copy the pre-edit `test_indices_generate_valid_indexers` over to our Numpy tests.

Reply:
Did you mean `True`? I wrote a new test `numpy/test_gen_data.py::test_basic_indices_can_generate_indices_not_covering_all_dims` that checks flat indices are indeed actually generated, which parallels what `array_api/test_indices.py/test_indices_generate_valid_indexers` does to check they're _not_ generated.

So I think in a logical sense I'm now covering both scenarios, but coverage is still failing. This has been a good excuse to finally start learning how to use coverage but yea haven't got round to fixing this today—let me know any quick fixes otherwise I'll explore this tomorrow.

Reply:
My mistake! I *did* mean `False`, even though there obviously are tests that execute it that way... they're just not run under coverage!

So slapping a `# pragma: no branch` on line 688 will fix this.